deeppavlov/dataset_readers/md_yaml_dialogs_reader.py

```diff
@@ -241,78 +246,142 @@ def _read_story(cls,
 
         curr_story_title = None
         curr_story_utters_batch = []
-        curr_story_bad = False
-        for line in open(story_fpath):
+        nonlocal_curr_story_bad = False  # can be modified as a nonlocal variable
+
+        def process_user_utter(line: str) -> List[List[Dict[str, Any]]]:
+            """
+            given the stories.md user line, returns the batch of all the dstc2 ways to represent it
+            Args:
+                line: the system line to generate dstc2 versions for
+
+            Returns:
+                all the possible dstc2 versions of the passed story line
+            """
+            nonlocal intent2slots2text, slot_name2text2value, curr_story_utters_batch, nonlocal_curr_story_bad
+            try:
+                possible_user_utters = cls.augment_user_turn(intent2slots2text, line, slot_name2text2value)
+                # dialogs MUST start with system replics
+                for curr_story_utters in curr_story_utters_batch:
+                    if not curr_story_utters:
+                        curr_story_utters.append(default_system_start)
+
+                utters_to_append_batch = []
+                for user_utter in possible_user_utters:
+                    utters_to_append_batch.append([user_utter])
+
+            except KeyError as e:
```

Comment:
```suggestion
            except KeyError:
```
you don't use KeyError object